bgpd/bgp_label.h

```diff
@@ -117,4 +117,7 @@ struct attr *bgp_labels_set(struct attr *attr,
 			    uint32_t num_labels, bool direct_copy,
 			    bool set_valid_label);
 
+#define bgp_labels_initialise(_attr_)                                          \
+	bgp_labels_set((_attr_), NULL, 0, true, false)
```

Comment:
This `define` adds a lot of complexity.

Do not use a `define` when a function is sufficient.

`bgp_labels_initialise` is a misleading name because it is only about VNI labels.

```
void bgp_attr_vni_labels_init(struct attr *attr)
{
	attr->num_labels = 0;
	for (i = 0; i < BGP_MAX_LABELS; i++)
		attr->label_tbl[i] = MPLS_INVALID_LABEL;
} 
```
use this function in `setlabels()` instead

Furthermore, you are removing some codes that you added in previous commits. Do that in the initial commits please

Reply:
see 

> Tests are mandatory to reflect what's going here.

done